constellation-app/constellation
# Enrichment/restore default dataview categories #1919

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

As mentioned in issue #1919 here is a current feature that allows the user to change the visible categories in the data view and change the order of the categories however resetting this takes some effort.  This enrichment simply adds a button, "Restore to Default",  to reset the dataview categories to the default which is all categories in the visible list in the default order.

![image](https://github.com/constellation-app/constellation/assets/63685177/393460da-4131-46b7-9dd7-2fa7a1a0515d)

### Alternate Designs

None

### Why Should This Be In Core?

This is a small change that will make the user experience better by saving time restoring to the defaults.

### Benefits

Saving time for the user to restore the categories to the default settings.

### Possible Drawbacks

None

### Verification Process

1. Open Setup > Options. Click the Constellation tab, then data access view tab.
2. Change the order of the categories and change the visible and hidden categories.
3. Click 'Restore to Default' 
4. Check the hidden categories should be empty and the visible categories should include all categories in the default order.
5. Close and Open Constellation to ensure the changes have persisted.
### Applicable Issues

#1919
